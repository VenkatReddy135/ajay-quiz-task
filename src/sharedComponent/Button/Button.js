/* eslint-disable react/destructuring-assignment */
import React from "react";
import PropTypes from "prop-types";

export default function Button(props) {
  return (
    <button
      type="button"
      onClick={props.handleClick}
      className={props.buttonClass}
      value={props.buttonIndex}
    >
      {props.buttonValue}
    </button>
  );
}

Button.propTypes = {
  buttonClass: PropTypes.string,
  handleClick: PropTypes.func,
  buttonValue: PropTypes.string,
  buttonIndex: PropTypes.any,
};

Button.defaultProps = {
  buttonIndex: "",
  buttonValue: "",
};
