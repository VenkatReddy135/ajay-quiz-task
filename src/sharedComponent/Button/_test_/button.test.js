/* eslint-disable testing-library/no-node-access */
import React from "react";
import { cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Button from "../Button";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

describe("Button  component", () => {
  const renderedData = renderer.create(<Button />).toJSON();

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button />, div);
  });

  it("renders a snapshot", () => {
    expect(renderedData).toMatchSnapshot();
  });

  afterEach(() => {
    cleanup();
  });
});
