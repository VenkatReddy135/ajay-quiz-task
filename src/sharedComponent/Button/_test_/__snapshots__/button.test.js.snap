// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  component renders a snapshot 1`] = `
<button
  type="button"
  value=""
>
  
</button>
`;
